src/harness/BenchmarkDotNet.Extensions/ValuesGenerator.cs

```diff
@@ -128,6 +128,8 @@ public static string[] ArrayOfStrings(int count, int minLength, int maxLength)
 
         private static T GenerateValue<T>(Random random)
         {
+            if (typeof(T) == typeof(byte))
+                return (T)(object)(byte)random.Next(byte.MinValue, byte.MaxValue);
```

Comment:
this method was missing `byte` support on purpose. Most of its callers don't support `byte` as they try to generate collections of unique values:

https://github.com/dotnet/performance/blob/3e63b12c2cfe8d3f4f8b6083e49ab28b5111cedd/src/harness/BenchmarkDotNet.Extensions/ValuesGenerator.cs#L26

Since `byte` gives us only 256 unique values, we can't generate a collection of more than 256 unique bytes and loops like this:

https://github.com/dotnet/performance/blob/3e63b12c2cfe8d3f4f8b6083e49ab28b5111cedd/src/harness/BenchmarkDotNet.Extensions/ValuesGenerator.cs#L106-L112

are never going to finish.

You should either remove the byte support or modify all the callers to throw when a unique collection of more than 256 bytes was requested to be created (my preffered solution).

Reply:
Will add the exception throw

Reply:
Done